ra/registration-authority.go

```diff
@@ -602,6 +608,12 @@ func (ra *RegistrationAuthorityImpl) UpdateRegistration(base core.Registration,
 
 // UpdateAuthorization updates an authorization with new values.
 func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization, challengeIndex int, response core.Challenge) (authz core.Authorization, err error) {
+	// Refuse to update expired authorizations
+	if base.Expires == nil || base.Expires.Before(ra.clk.Now()) {
+		err = fmt.Errorf("Expired authorization")
```

Comment:
This will cause us to return a 500 back to the user, no?

I think we need to make this a real error case and weave it through the RPC layer to get a 4xx back to them.